packages/blade/src/utils/isPartialMatchObjectKeys/isPartialMatchObjectKeys.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable no-console */
+import isObject from 'lodash/isObject';
+
+export type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Record<number | string, unknown> ? DeepPartial<T[P]> : T[P];
+};
+
+const isPartialMatchObjectKeys = <ActualObject>({
+  objectToMatch,
+  objectToInspect,
+}: {
+  objectToMatch: DeepPartial<ActualObject>;
+  objectToInspect: ActualObject;
+}): boolean => {
+  const matchResponses: boolean[] = [];
+
+  const matchObjectKeys = ({
+    innerObjectToMatch,
+    innerObjectToInspect,
+  }: {
+    innerObjectToMatch: DeepPartial<ActualObject>;
+    innerObjectToInspect: DeepPartial<ActualObject>;
+  }): void => {
+    for (const [key, valueToMatch] of Object.entries(innerObjectToMatch)) {
+      const valueToInspect = innerObjectToInspect[key as keyof DeepPartial<ActualObject>];
+
+      if (innerObjectToInspect.hasOwnProperty(key)) {
+        if (
+          valueToMatch === null ||
+          valueToMatch === undefined ||
+          valueToMatch === '' ||
+          Array.isArray(valueToMatch) ||
+          (!(valueToMatch instanceof Object) && typeof valueToMatch !== typeof valueToInspect)
```

Comment:
Can we simplify the conditional?

Reply:
any suggestions? The reason is all the other conditions I can test without negation but not getting any simpler way for this one. The idea here is that I want to check if it's not instance of an object then the typeof **match** should be equal to type of **inspect** and if it's not then it's a failure case.

Maybe I'll add a comment to make it easier to read?

Reply:
Okay, comment should be fine for now I think.